CenterForOpenScience/osf.io
# [ENG-2469] Upgrade pip

## Purpose

We wanted to upgrade our version of pip and found that the conflict resolver did not work as expected, so I decide to resolve our dependency conflicts bey updating our conflicting dependencies. One of those dependencies is pytest and pytest 5.0.0 outlawed certain fixture behavior, so I had to change some of those fixtures to functions. 

The testmon PR has a lot of overlap with this, because that also upgrades the conflicting dependencies.

## Changes

- updates some old dependencies to update pip and alpine. 

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify pip installs correctly and the site deploys well.

What are the areas of risk?

We are making changes to our Dockerfile.


## Documentation



tech task, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2469
